Support for OPT Switch command for AW-HE40 (and AW-HE38)

CountParadox opened the pull request:
The OPT switch is used to turn an IR Cut filter on and off within the camera by physically moving it in front of the sensor. In normal state this prevents IR light from hitting the sensor and allows for normal colour video, when this is turned off it allows for IR illumination such as IR spotlights and floodlights, and the picture turns to black and white.

I am not a good programmer but I have done my best to implement this into this pull request!

CountParadox commented:
This may work on many of the other cameras, but I do not have them to test and confirm.

CountParadox commented:
Command pulled from https://eww.pass.panasonic.co.jp/pro-av/support/content/guide/DEF/HE50_120_SERIAL/ConvertibleProtocol.pdf

premultiply commented:
Night shot mode support is already included in https://github.com/bitfocus/companion-module-panasonic-cameras

CountParadox commented:
@premultiply I cannot find how to enable it? What is it called ? Is that perhaps seperate to this ? The HE38/40 have physical IR filter that gets moved from the sensor

premultiply commented:
https://github.com/bitfocus/companion-module-panasonic-cameras/blob/2a6cdf98ee674efebeb02e1a2da59092b8683244/src/actions.js#L441-L449

CountParadox commented:
right, so the only problem is that HE38/40 are not listed as supporting that ?

CountParadox commented:
I have updated the files to remove my addition of opt, and adjusted to support night for the HE40/38, is this better ? I cannot see *any* of the cameras having night as supported, so maybe thats why I couldnt find it?

CountParadox commented:
oh! These are two seperate modules!

CountParadox commented:
How do I add the new module to my companion? It does not show up for me.

premultiply commented:
(Update Companion and) add a new device connection of type "Panasonic Camera".

CountParadox commented:
@premultiply sorry, to clarify, its only available in the beta at this time? because running the latest version 3.3.1+7001 I am not getting the option for Panasonic Camera, only the legacy PTZ Camera one!

premultiply commented:
Yes, use the beta builds.
